common/delete-link.js

```diff
@@ -47,12 +48,14 @@
             link: function(scope) {
                 scope.deleteFn = function deleteFn() {
```

Comment:
More likely that this whole function should be wrapped in the wrapper rather than wrapping the callback.